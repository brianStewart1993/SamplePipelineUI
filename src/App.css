#app .logout-btn{
  float: right;
  margin-left: auto;

}

#app .logout-btn a{
  text-decoration: none;
  color: #fff;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}



@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
