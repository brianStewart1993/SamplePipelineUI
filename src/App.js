import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import LoginForm from './Components/Login/Login';
import SignUp from './Components/SignUp/SignUp';
import UserList from './Components/User/UserList';
import UserEdit from './Components/User/UserEdit';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { shallow } from 'enzyme';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

/*import { Router, Route, hashHistory } from 'react-router'*/

/*const RootStack = createStackNavigator({
  Home: {
    screen: App
  }
});*/

class App extends Component {




  constructor(props){
    super(props);
    /*this.handleUsernameChange = this.handleUsernameChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);*/

    this.state = {
      username: "",
      password: "",
      users: []
    };


  }






  render() {
    /*return <Login />;*/
    return (
      <Router>

        <div id="app">
          <AppBar position="static" color="primary">
            <Toolbar>
              <Typography  variant="title" color="inherit">
                Cash Secured Portal
              </Typography>

                <Button color="inherit" className="logout-btn"  >
                  <Link to="/">
                    Logout
                  </Link>
                </Button>

            </Toolbar>
          </AppBar>
          <Route exact path="/" component={LoginForm} />
          <Route path="/sign-up" component={SignUp} />
          <Route path="/user-list" component={UserList} />
          <Route path="/edit-user/:id" component={UserEdit} />

        </div>
      </Router>
    );
  }


}


export default App;
