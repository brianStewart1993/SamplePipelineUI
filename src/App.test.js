import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Renders correct layout', () => {
  /*const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();*/

  // when
  const component = shallow(<App />);
  // then
  expect(component.getElements()).toMatchSnapshot();
});
