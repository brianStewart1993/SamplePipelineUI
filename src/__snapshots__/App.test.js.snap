// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correct layout 1`] = `
Array [
  <BrowserRouter>
    <div
        id="app"
    >
        <WithStyles(AppBar)
            color="primary"
            position="static"
        >
            <WithStyles(Toolbar)>
                <WithStyles(Typography)
                    color="inherit"
                    variant="title"
                >
                    Cash Secured Portal
                </WithStyles(Typography)>
                <WithStyles(Button)
                    className="logout-btn"
                    color="inherit"
                >
                    <Link
                        replace={false}
                        to="/"
                    >
                        Logout
                    </Link>
                </WithStyles(Button)>
            </WithStyles(Toolbar)>
        </WithStyles(AppBar)>
        <Route
            component={[Function]}
            exact={true}
            path="/"
        />
        <Route
            component={[Function]}
            path="/sign-up"
        />
        <Route
            component={[Function]}
            path="/user-list"
        />
        <Route
            component={[Function]}
            path="/edit-user/:id"
        />
    </div>
</BrowserRouter>,
]
`;
